### DISCUSSION TITLE
Use PInvoke

### DISCUSSION THREAD

**Discussion by NN---** [dahall/Vanara#197] [Ideas]
MS announced official support for C# pinvoke based on automatic tooling.
Does it make sense to use this tooling or the PInvoke itself in Vanara making Vanara simply higher level API above the low level?
https://blogs.windows.com/windowsdeveloper/2021/01/21/making-win32-apis-more-accessible-to-more-languages/

**Comment by dahall** [comment]
Thanks for pointing this out. I wasn't aware of it. I have followed [AArnott's project](https://github.com/dotnet/pinvoke) and even thought about combining work with that project. In the end, I decided that I didn't like the bias towards unsafe code and pointers and the project's decision to only support the latest version of .NET. As I dig more into [win32metadata](https://github.com/microsoft/win32metadata) and [C#/Win32](https://github.com/microsoft/CsWin32) I'll update this post with the direction I decide to take the project.

**Comment by NN---** [comment]
The post is from today. Nobody was aware of it ;)
